lib/util/html.js

```diff
@@ -15,24 +15,22 @@ const envDomain = config.get('default.env.domain');
 
 // Logic adapted from `login.vue` in frontend.
 const safeNextPathFrom = next => {
-  if (!next) return '/#/';
+  if (!next) return '/';
 
   let url;
   try {
     url = new URL(next, envDomain);
   } catch (e) {
-    return '/#/';
+    return '/';
   }
 
   if (url.origin !== envDomain || url.pathname === '/login')
-    return '/#/';
+    return '/';
 
   // Don't modify enketo URLs
   if (url.pathname.startsWith('/-/')) return url.toString();
```

Comment:
Don't know why we have to return full URL for Enketo including origin

Reply:
`url.toString()` seems simpler than `url.pathname + url.search + url.hash`

I suspect the specific reason was per the comment on L16:

```js
// Logic adapted from `login.vue` in frontend.
```

In frontend (https://github.com/getodk/central-frontend/blob/eb602bd563d41f560fa59dbb1ccac624b977df92/src/components/account/login.vue#L162-L163):

```js
      if (url.pathname.startsWith(`${enketoBasePath}/`))
        return external(url.href);
```

Why `.toString()` is used in preference to `.href` is unclear, but they appear to be equivalent per the spec (discussion at https://stackoverflow.com/a/54380688).

Reply:
In that file in Frontend, `external()` is basically just `window.location.replace()`. Maybe `url.pathname + url.search + url.hash` would have worked as well, but by that point in Frontend, it's already been validated that the URL is at the correct origin. We do use `url.pathname + url.search + url.hash` on the next line, in the default `return`:

```js
return internal(url.pathname + url.search + url.hash);
```

That's needed because `internal()` is hitting Vue Router, which just wants a path, not the full URL.

Reply:
Another thing I noticed is that if you try to navigate to an Enketo URL that requires auth, it redirects you to login in Frontend. It sets the `next` query parameter equal to the full Enketo URL. Whereas when Frontend sets the `next` query parameter for a Frontend page that requires auth, the value of the query parameter is just the path, not the full URL.

Reply:
The only place I see `safeNextPathFrom()` being used is to generate a `nextPath` to pass to `loaderTemplate`. Looking at `loaderTemplate`, I don't think that a distinction needs to be made between Enketo paths and Frontend paths. It's using `nextPath` in pretty generic ways. Maybe it'd be simpler for `safeNextPathFrom()` to do the same thing for the two types of paths: the previous logic had to account for the hash in Frontend URLs, but not anymore. No strong feelings though.